test/instrumentation/modules/@aws-sdk/s3-v3.test.js

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+if (process.env.GITHUB_ACTIONS === 'true' && process.platform === 'win32') {
+  console.log('# SKIP: GH Actions do not support docker services on Windows')
+  process.exit(0)
+}
+
```

Comment:
```suggestion
if (process.env.ELASTIC_APM_CONTEXT_MANAGER === 'patch') {
  console.log('# SKIP @aws-sdk/* instrumentation does not work with contextManager="patch"')
  process.exit()
}
```

Because `async/await` is involved in the instrumentation there is no way it can work with the (now deprecated) `contextManager: "patch"` config option (see https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html#context-manager for some details).  We can just skip the tests for this case.   This is why the CI test jobs with ", patch" in the name are failing.

Reply:
Thanks for this pearl of wisdom. I need to do a deep dive on this topic :)